### DISCUSSION TITLE
a new idea for autoclose

### DISCUSSION THREAD

**Discussion by EliasA5** [nvim-tree/nvim-tree.lua#2021] [Show and tell]
Added the following 2 lines to my remaps:
`vim.keymap.set("n", ":q<CR>", ":NvimTreeClose<CR>:q<CR>", { nowait = true })`
`vim.keymap.set("n", ":", ":", { nowait = true })`
The first one simply closes the nvimtree first then reissues the `:q` command.
The second one purpose that if you start typing `:` it will show in the command line, despite starting to match the first command.

I can't think of any problems that this would cause, and it only acts after typing the full command.

**Comment by alex-courtis** [comment]
Please add to the wiki: https://github.com/nvim-tree/nvim-tree.lua/wiki/Auto-Close

There are many ways to auto close; it's good to have as many options as possible.

**Reply by EliasA5** [reply]
Actually I found out that method is unreliable and interferes with some plugins like which-key.
In the Autoclose wiki they only used the 'BufEnter' event, why not use the 'QuitPre' event; it only works when you issue the :q command and not when the NvimTree is the last buffer open, that is more intuitive, atleast for me.
a simple working example:
```lua
vim.api.nvim_create_autocmd({"QuitPre"}, {
    callback = function() vim.cmd("NvimTreeClose")  end,
})
```

**Reply by alex-courtis** [reply]
> Actually I found out that method is unreliable and interferes with some plugins like which-key.

That does not surprise me at all.

> QuitPre

That's quite nice, thanks for updating.
